examples/thermostat/silabs/src/SensorManager.cpp

```diff
@@ -109,18 +111,20 @@ void SensorManager::SensorTimerEventHandler(void * arg)
         simulatedIndex++;
         nbOfRepetition = 0;
     }
-#endif // USE_TEMP_SENSOR
+#endif // defined(SL_MATTER_USE_SI70XX_SENSOR) && SL_MATTER_USE_SI70XX_SENSOR
 
     SILABS_LOG("Sensor Temp is : %d", temperature);
 
+    MarkAttributeDirty reportState = MarkAttributeDirty::kNo;
     if ((temperature >= (lastTemperature + kMinTemperatureDelta)) || temperature <= (lastTemperature - kMinTemperatureDelta))
     {
-        lastTemperature = temperature;
-        PlatformMgr().LockChipStack();
-        // The SensorMagager shouldn't be aware of the Endpoint ID TODO Fix this.
-        // TODO Per Spec we should also apply the Offset stored in the same cluster before saving the temp
-
-        app::Clusters::Thermostat::Attributes::LocalTemperature::Set(kThermostatEndpoint, temperature);
-        PlatformMgr().UnlockChipStack();
+        reportState = MarkAttributeDirty::kYes;
```

Comment:
should it be kIfChanged to compared to the last report

Reply:
No, since we check that the value differs of a certain threshold. If we get here, we know the value changed for sure.
kYes and kIfChanged would have the same behavior.